articles/sorting-algorithms-in-js/index.md

```diff
@@ -0,0 +1,508 @@
+
+- **Average case time complexity**: It is a function defined as a result of the average number of steps taken on any instance of size n. It is usually expressed in [Big theta notation](https://www.geeksforgeeks.org/analysis-of-algorithms-set-3asymptotic-notations/).
+
+- **Best case time complexity**: It is a function defined as a result of the minimum number of steps taken on any instance of size n. It is usually expressed in [Big omega notation](https://www.geeksforgeeks.org/analysis-of-algorithms-set-3asymptotic-notations/).
+
+- **Space complexity**: It is a function defined as a result of additional memory space needed to carry out the algorithm. It is usually expressed in [Big O notation](https://www.geeksforgeeks.org/analysis-of-algorithms-set-3asymptotic-notations/).
+
+### Strategies applied during sorting
+
+- **Recursion**: The algorithm accomplishes its task by repeating a particular process in a particular order.
+
+- **Divide and conquer**: The algorithm accomplished its task by dividing the data into smaller portions so that to handle it.
+
+### Sorting algorithms
+
+For each sorting algorithm discussed below, there is: a step-by-step explanation of how the algorithm works, pictorial representation, and implementation of the algorithm using JavaScript. 
+
+### Bubble sort.
+
+Bubble sort follows the recursion technique.
+
+**Step-by-step guide**:
+
+- Start by comparing two elements in a list.
+
+- Swap them if required.
+
+- Continue till the end of the list. At this point, you have made a series of inner passes and completed an outer pass.
+
+- Repeat the process until the entire list is sorted.
+
+**Pictorial representation**:
+
+![bubble_sort](/engineering-education/sorting-algorithms-in-js/bubble-sort.png)
+
+**JavaScript implementation**.
+
+```JavaScript
+function bubbleSort(arr){
+
+    //Outer pass
+    for(let i = 0; i < arr.length; i++){
+
+        //Inner pass
+        for(let j = 0; j < arr.length - i - 1; j++){
+
+            //Value comparison using ascending order
+            
+            if(arr[j + 1] < arr[j]){
+
+                //Swapping
+                [arr[j + 1],arr[j]] = [arr[j],arr[j + 1]]
+            }
+        }
+    };
+    return arr;
+};
+
+console.log(bubbleSort([5,3,8,4,6]));
+```
+
+Output
+
+```bash
+[3,4,5,6,8]
+```
+
+Bubble sort has the following performance cases:
+
+- Worst-case time complexity of [`Big O`](https://www.geeksforgeeks.org/difference-between-big-oh-big-omega-and-big-theta/) n ^2. Requires n steps to sort a list of n elements.
+
+- Average-case time complexity of [`Big theta`](https://www.geeksforgeeks.org/difference-between-big-oh-big-omega-and-big-theta/) n ^ 2. Requires n steps to sort a list of n elements.
+
+- Best-case time complexity of [`Big omega`](https://www.geeksforgeeks.org/difference-between-big-oh-big-omega-and-big-theta/): O(n). Requires n steps to sort a sorted list of n elements.
```

Comment:
You don't have to provide the same links everywhere. Just provide it at the intro where you are talking about the notations.

Reply:
fixed.